services/121-service/src/program-financial-service-provider-configurations/program-financial-service-provider-configurations.repository.ts

```diff
@@ -16,47 +20,119 @@ export class ProgramFinancialServiceProviderConfigurationRepository extends Repo
+  public async findUsernamePasswordPropertiesForIds(
+    programFinancialServiceProviderConfigurationId: number[],
+  ): Promise<
+    {
+      programFinancialServiceProviderConfigurationId: number;
+      credentials: UsernamePasswordInterface;
+    }[]
+  > {
+    return await Promise.all(
+      programFinancialServiceProviderConfigurationId.map(async (id) => ({
+        programFinancialServiceProviderConfigurationId: id,
+        credentials: await this.getUsernamePasswordProperties(id),
+      })),
+    );
+  }
+
+  public async getUsernamePasswordProperties(
+    programFinancialServiceProviderConfigurationId: number,
+  ): Promise<UsernamePasswordInterface> {
+    const properties = await this.getConfigurationProperties(
+      programFinancialServiceProviderConfigurationId,
+    );
+    const propertyUsername = properties.find(
+      (c) =>
+        c.name === FinancialServiceProviderConfigurationProperties.username,
+    );
+    const propertyPassword = properties.find(
+      (c) =>
+        c.name === FinancialServiceProviderConfigurationProperties.password,
+    );
+
+    const response: UsernamePasswordInterface = {
+      username: null,
+      password: null,
+    };
+
+    if (typeof propertyUsername?.value == 'string') {
+      response.username = propertyUsername.value;
+    }
+    if (typeof propertyPassword?.value == 'string') {
+      response.password = propertyPassword.value;
+    }
+    return response;
+  }
+
+  public async getPropertiesByNamesOrThrow({
+    programFinancialServiceProviderConfigurationId,
     names,
   }: {
-    programId: number;
-    financialServiceProviderName: FinancialServiceProviderName;
+    programFinancialServiceProviderConfigurationId: number;
     names: string[];
   }) {
-    const configurations = await this.baseRepository.find({
-      where: {
-        programId: Equal(programId),
-        fsp: { fsp: Equal(financialServiceProviderName) },
-        name: In(names),
-      },
-    });
+    const properties = await this.getConfigurationProperties(
+      programFinancialServiceProviderConfigurationId,
+    );
+
     for (const name of names) {
-      if (
-        !configurations.find((configuration) => configuration.name === name)
-      ) {
+      if (!properties.find((property) => property.name === name)) {
         throw new Error(
-          `Configuration with name ${name} not found for program ${programId} and FSP ${financialServiceProviderName}`,
+          `Configuration with name ${name} not found for ProgramFinancialServiceProviderConfigurationEntity with id:  ${programFinancialServiceProviderConfigurationId}`,
         );
       }
     }
 
-    return configurations.map((configuration) => ({
-      name: configuration.name,
-      value: configuration.value,
+    return properties.map((property) => ({
+      name: property.name,
+      value: property.value,
     }));
   }
+
+  public async getPropertyValueByNameOrThrow({
+    programFinancialServiceProviderConfigurationId,
```

Comment:
I think it may be clearer to leave it like this. Because people may be confused if it is programFinancialServiceProviderConfigurationId or programFinancialServiceProviderConfigurationPropertyId.